sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -360,16 +364,13 @@ def translate_output_properties(res: 'Resource', output: Any) -> Any:
         log.debug(f"incoming output value translated: {value} -> {translated_value}")
         all_properties[translated_key] = translated_value
 
-    for key, value in props.items():
-        if key not in all_properties:
+    for key, value in list(serialized_props.items()):
+        translated_key = res.translate_output_property(key)
+        if translated_key not in all_properties:
             # input prop the engine didn't give us a final value for.  Just use the value passed into the resource
             # after round-tripping it through serialization. We do the round-tripping primarily s.t. we ensure that
             # Output values are handled properly w.r.t. unknowns.
-            input_prop = await serialize_property(value, [])
-            if input_prop is None:
-                continue
-
-            all_properties[key] = deserialize_property(input_prop)
+            all_properties[translated_key] = translate_output_properties(res, deserialize_property(value))
```

Comment:
FWIW - The comment here is a little more confusing now because the "roundtripping" is now split between the serialization we do to prepare the RCP and the deserialization we do here.  It feels much less like round-tripping to me now.

Maybe something like:

```
# if the engine did not provide a value for an output which we had as an input use the value that
# was passed in by the user when they registered the resource.
```

Or something?

Reply:
Sure, that's fine.